test/driver.cpp

```diff
@@ -81,32 +82,23 @@ int main(int argc, char* argv[])
 }
 
 
-//declare suite runners: one for each test suite
-//also add an entry in the "suites" map for each suite defined
-void array_test();
-
-
+//run test suites in sequence: runs all suites defined or only those specified on cmd-line
 void run_suites(const Options& options)
 {
-	using suite_runner_type = void(*)();
-	std::unordered_map<std::string, suite_runner_type> suites{
-		//add one entry for each test suite to run
-		//suite-name and suite-runner, both in braces
-		//not necessary, but recommend using the suite_runner function's name as the suite name
-		{"array_test", array_test}
-	};
-
-	//build a collection of suites to run if necessary
+	//retrieve all test suites defined
+	auto suites = get_test_suites();
```

Comment:
Would it make more sense to use uniform initialization syntax here (and in the lines below)?

Reply:
I believe I mentioned this in response to an earlier comment, but I expand on it here:

There is no benefit to using UIS with auto typed data.  By definition, auto chooses the exact type of the initializer and thus using UIS is pointless.

In general, people don't like UIS because it causes visual clutter and obscures the initial value, and makes it especially hard to comprehend what is going on when the initializer is a non-trivial expression. Thus, we should use UIS where we can't (and shouldn't) use auto.